### PR TITLE
Add ImageFilter and BackdropFilter to CanvasKit backend

### PR DESCRIPTION
**Pull request by hterkelsen**
This shows how it looks with a Cupertino dialog (ignore ugly fonts):

backdropfilter

### CODE REVIEW

File: lib/web_ui/lib/src/engine/compositor/image_filter.dart [hunk 1]
```diff
@@ -0,0 +1,27 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+part of engine;
+
```
Comment (nturgut) [inline lib/web_ui/lib/src/engine/compositor/image_filter.dart]: Can we add comments to the class?

File: lib/web_ui/lib/src/ui/painting.dart [hunk 2]
```diff
@@ -1055,7 +1055,7 @@ class Paint {
       _paintData = _paintData.clone();
       _frozen = false;
     }
-   _paintData.color = value.runtimeType == Color ? value : Color(value.value);
+    _paintData.color = value.runtimeType == Color ? value : Color(value.value);
```
Comment (nturgut) [inline lib/web_ui/lib/src/ui/painting.dart]: nit: extra space

File: lib/web_ui/lib/src/engine/compositor/util.dart [hunk 3]
```diff
@@ -156,6 +156,12 @@ js.JsObject makeSkPaint(ui.Paint paint) {
     skPaint.callMethod('setMaskFilter', <js.JsObject>[skMaskFilter]);
   }
 
+  if (paint.imageFilter != null) {
+    final SkImageFilter skImageFilter = paint.imageFilter;
+    skPaint.callMethod(
+        'setImageFilter', <js.JsObject>[skImageFilter.skImageFilter]);
+  }
+
```
Comment (nturgut) [inline lib/web_ui/lib/src/engine/compositor/util.dart]: question: does the order of method calls make a difference? If so may be we should document it.

File: lib/web_ui/lib/src/engine/compositor/image_filter.dart [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+        sigmaX,
+        sigmaY,
+        canvasKit['TileMode']['Clamp'],
+        null,
+      ],
+    );
+  }
+
+  double get sigmaX => throw UnimplementedError(
+      'CanvasKit backend doesn\'t support ImageFilter.sigmaX');
```
Comment (yjbanov) [inline lib/web_ui/lib/src/engine/compositor/image_filter.dart]: We shouldn't have these in `ui`. Can you please move them under `src/engine`?

File: lib/web_ui/lib/src/engine/compositor/initialization.dart [hunk 5]
```diff
@@ -9,7 +9,7 @@ const bool experimentalUseSkia =
     bool.fromEnvironment('FLUTTER_WEB_USE_SKIA', defaultValue: false);
 
 /// The URL to use when downloading the CanvasKit script and associated wasm.
-const String canvasKitBaseUrl = 'https://unpkg.com/canvaskit-wasm@0.7.0/bin/';
+const String canvasKitBaseUrl = 'https://particles.skia.org/static/';
```
Comment (yjbanov) [inline lib/web_ui/lib/src/engine/compositor/initialization.dart]: Will this make our tests unstable? Maybe host our own copy on a firebase static site? This way we can control when to upload a new version.

File: lib/web_ui/lib/src/ui/painting.dart (hunk 2 above)
Comment (hterkelsen) [reply lib/web_ui/lib/src/ui/painting.dart]: it was incorrectly indented before

File: lib/web_ui/lib/src/engine/compositor/image_filter.dart (hunk 1 above)
Comment (hterkelsen) [reply lib/web_ui/lib/src/engine/compositor/image_filter.dart]: Done.

File: lib/web_ui/lib/src/engine/compositor/util.dart (hunk 3 above)
Comment (hterkelsen) [reply lib/web_ui/lib/src/engine/compositor/util.dart]: Order doesn't matter

File: lib/web_ui/lib/src/engine/compositor/image_filter.dart (hunk 4 above)
Comment (hterkelsen) [reply lib/web_ui/lib/src/engine/compositor/image_filter.dart]: Done.

File: lib/web_ui/lib/src/engine/compositor/initialization.dart (hunk 5 above)
Comment (hterkelsen) [reply lib/web_ui/lib/src/engine/compositor/initialization.dart]: There are currently no CanvasKit tests. We can worry about this when we enable testing for CanvasKit

Comment (yjbanov) [reply lib/web_ui/lib/src/engine/compositor/initialization.dart]: Ah right, one more thing to add to our testing strategy :)

**Pull request merged**